rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,364 @@
+
+## User-facing explanation
+
+The [bevy-ui-navigation README][`bevy-ui-navigation`] is a good start.
+To summarize, here are the headlines:
+- The navigation system can only be interacted with with events:
+  `NavRequest` as input and `NavEvent` as output.
+- `NavEvent::FocusChanged` contains the list of entities that are traversed
+  to go from the last focus position to the new one.
+- ui-navigation provides default input systems, 
+  but it's possible to disable them and instead use custom ones.
+  This is why we restrict interactions with navigation to events.
+- ui-navigation provides a system to declare buttons as "leading to menu X"
+  in order to create menu trees.
+- ui-navigation defines [custom `SystemParam`][event_helpers]s
+  to ease reaction to events (such a button press or menu change).
+  The most naive usage results into [elm architecture][elm_architecture]-like code.
+  Systems updating the model (ECS hierarchy) and systems reacting to events. 
+- It is possible to create isolated menus by using the `NavMenu` component.
+- All `Focusable` children in the `NavMenu` entity's tree will be part
+  of this menu, and moving using a gamepad from one focusable in a menu
+  can only lead to another focusable within the same menu.
+  You can specify a focusable in any other menu that will lead to this
+  one once activated.
+
+## Implementation
+
+Since `bevy-ui-navigation` already exists,
+we will discuss its design, it's functionalities and their uses.
+
+Implementation details are already covered by the implementation,
+which btw is fully documented (including internal architecture).
+
+### Exposed API
+
+The crux of `bevy-ui-navigation` are the following types:
+- `Focusable` component
+  - `lock`: activating it will lock the navigation engine and discard all `NavRequest`.
+  - `cancel`: activating it will do the equivalent of receiving a `Cancel` request.
+  - `dormant`: this will be the preferred entity to focus when entering a menu.
+- `NavMenu` enum
+  - `Wrapping**`/`Bound**`: navigation can wrap
+    if directional sticks are pressed in a direction there are no focusables in
+  - `**Scope`/`**2d`: A scopped menu is tab menu navigable with special hotkeys.
+- `NavRequest` event
+  - `Action`: equivalent to left click or `A` button on controller
+  - `Cancel`: backspace or `B` on controller
+  - `FocusOn`: request to move the focus to a specific `Entity`
+  - `Move(Direction)`: directional gamepad or keyboard input
+  - `ScopeMove(ScopeDirection)`: tab menu hotkey
+- `NavEvent` event
+  - `InitiallyFocused`: We went from 0 focused to 1 focused element
+  - `NoChanges`: The `NavRequest` did not change the focus state.
+  - `FocusChanged`: The `NavRequest` changed the focus state,
+    has the list of items that were made inactive
+    and items made active.
+  - `Locked`: The `NavRequest` caused the focus system to lock.
+  - `Unlocked`: Something unlocked the focus system.
+- `MoveParam` trait
```

Comment:
This is a odd name. Maybe something like `MenuNavigationStrategy`? It's a niche trait so we can go with something verbose and descriptive.

Reply:
I love this suggestion. This is a very good name